vms/platformvm/camino_service.go

```diff
@@ -63,6 +85,135 @@ type GetConfigurationReply struct {
 	CodecVersion json.Uint16 `json:"codecVersion"`
 }
 
+// GetBalance gets the balance of an address
+func (service *CaminoService) GetBalance(_ *http.Request, args *GetBalanceRequest, response *GetBalanceResponseV2) error {
+	caminoGenesis, err := service.vm.state.CaminoGenesisState()
+	if err != nil {
+		return err
+	}
+	if !caminoGenesis.LockModeBondDeposit {
+		return service.Service.GetBalance(nil, args, &response.GetBalanceResponse)
+	}
+
+	if args.Address != nil {
+		args.Addresses = append(args.Addresses, *args.Address)
+	}
+
+	service.vm.ctx.Log.Debug("Platform: GetBalance called",
+		logging.UserStrings("addresses", args.Addresses),
+	)
+
+	// Parse to address
+	addrs, err := avax.ParseServiceAddresses(service.addrManager, args.Addresses)
+	if err != nil {
+		return err
+	}
+
+	utxos, err := avax.GetAllUTXOs(service.vm.state, addrs)
+	if err != nil {
+		return fmt.Errorf("couldn't get UTXO set of %v: %w", args.Addresses, err)
+	}
+
+	unlockedOutputs := map[ids.ID]uint64{}
+	bondedOutputs := map[ids.ID]uint64{}
+	depositedOutputs := map[ids.ID]uint64{}
+	depositedBondedOutputs := map[ids.ID]uint64{}
+
+utxoFor:
+	for _, utxo := range utxos {
+		assetID := utxo.AssetID()
+		switch out := utxo.Out.(type) {
+		case *secp256k1fx.TransferOutput:
+			newBalance, err := math.Add64(unlockedOutputs[assetID], out.Amount())
+			if err != nil {
+				unlockedOutputs[assetID] = stdmath.MaxUint64
+			} else {
+				unlockedOutputs[assetID] = newBalance
+			}
+		case *locked.Out:
+			switch out.LockState() {
+			case locked.StateBonded:
+				newBalance, err := math.Add64(bondedOutputs[assetID], out.Amount())
+				if err != nil {
+					bondedOutputs[assetID] = stdmath.MaxUint64
+				} else {
+					bondedOutputs[assetID] = newBalance
+				}
+			case locked.StateDeposited:
+				newBalance, err := math.Add64(depositedOutputs[assetID], out.Amount())
+				if err != nil {
+					depositedOutputs[assetID] = stdmath.MaxUint64
+				} else {
+					depositedOutputs[assetID] = newBalance
+				}
+			case locked.StateDepositedBonded:
+				newBalance, err := math.Add64(depositedBondedOutputs[assetID], out.Amount())
+				if err != nil {
+					depositedBondedOutputs[assetID] = stdmath.MaxUint64
+				} else {
+					depositedBondedOutputs[assetID] = newBalance
+				}
+			default:
+				service.vm.ctx.Log.Warn("Unexpected utxo lock state")
+				continue utxoFor
+			}
+		default:
+			service.vm.ctx.Log.Warn("unexpected output type in UTXO",
+				zap.String("type", fmt.Sprintf("%T", out)),
+			)
+			continue utxoFor
+		}
+
+		response.UTXOIDs = append(response.UTXOIDs, &utxo.UTXOID)
+	}
+
+	balances := map[ids.ID]uint64{}
+	for assetID, amount := range unlockedOutputs {
+		newBalance, err := math.Add64(balances[assetID], amount)
+		if err != nil {
+			balances[assetID] = stdmath.MaxUint64
```

Comment:
What the reasoning behind shadowing the error and setting a default value here?

Reply:
that stems from the avalanche implementation and thus I cannot answer about the reasoning behind this refactoring